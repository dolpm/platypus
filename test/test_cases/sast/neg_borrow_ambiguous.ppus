pipe test |'a, 'b| [x: &'a int, y: &'b int] |> &'b int {
  if (@x == 2) {
    |> x;
  }
  |> y;
}

pipe main [] |> unit {
  int x = 4;
  mut &int w = &x;
  while (x < 5) {
    int y = 4;
    loop 0 -> 5 as z {
      w = (test <| [&x, &y]);
    }
  }
  |> ();
}
