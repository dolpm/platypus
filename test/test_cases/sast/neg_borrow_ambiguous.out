Fatal error: exception Failure("the variable being bound is outlived by the definition to which it is being bound.")
